rust/src/statistics/scripts/verkle_node_plots.py

```diff
@@ -0,0 +1,149 @@
+    plt.show()
+
+
+def filter_and_group(df, kind, size_col="Node Size", count_col="Count", threshold=0.01):
+    """
+    Filter subtypes of a given kind by a minimum percentage threshold.
+    Returns a DataFrame with subtypes above the threshold.
+    """
+    sub_df = df[(df["Node Kind"] == kind) & (df[size_col] != kind)]
+    total = sub_df[count_col].sum()
+    return sub_df[sub_df[count_col] / total >= threshold], total
+
+
+def pie_with_other(df, kind, threshold, label_col="Node Size", count_col="Count"):
+    """
+    Pie plot for a node kind, grouping subtypes below threshold into 'Other'.
+    """
+    sub_df = df[(df["Node Kind"] == kind) & (df[label_col] != kind)].copy()
+    total = sub_df[count_col].sum()
+    sub_df["Percentage"] = sub_df[count_col] / total
+    above = sub_df[sub_df["Percentage"] >= threshold]
+    below = sub_df[sub_df["Percentage"] < threshold]
+    if not below.empty:
+        other_count = below[count_col].sum()
+        above = pd.concat(
+            [above, pd.DataFrame([{label_col: "Other", count_col: other_count}])],
+            ignore_index=True,
+        )
+    pie_plot(
+        above,
+        count_col,
+        label_col,
+        f"Distribution of {kind} node children ({threshold * 100:.1f}% threshold)",
+    )
+
+
+def set_plot_params():
+    plt.rcParams["axes.labelsize"] = 16
+    plt.rcParams["xtick.labelsize"] = 16
+    plt.rcParams["ytick.labelsize"] = 16
+    plt.rcParams["legend.fontsize"] = 16
+    plt.rcParams["axes.titlesize"] = 20
+
+
+# %%  --- Main analysis and plotting ---
+
+set_plot_params()
+df = load_and_prepare_data(CSV_PATH)
+
+# Pie: All node kinds (with totals)
+all_kinds_df = df[df["Node Kind"] == df["Node Size"]]
+pie_plot(all_kinds_df, "Count", "Node Kind", "Distribution of Node Kinds")
+
+# Pie: All node kinds, excluding "Empty"
+non_empty_df = all_kinds_df[all_kinds_df["Node Kind"] != "Empty"]
+pie_plot(
+    non_empty_df,
+    "Count",
+    "Node Kind",
+    "Distribution of Node Kinds (excluding Empty nodes)",
+)
+
+# Pie: Leaf node subtypes above threshold
+leaf_threshold = 0.01
+leaf_subtypes_df, _ = filter_and_group(df, "Leaf", threshold=leaf_threshold)
+pie_plot(
+    leaf_subtypes_df,
+    "Count",
+    "Node Size",
+    f"Distribution of Leaf node children ({leaf_threshold * 100:.2f}% threshold)",
+)
+
+# Pie: Inner node subtypes above threshold
+inner_threshold = 0.01
+inner_subtypes_df, _ = filter_and_group(df, "Inner", threshold=inner_threshold)
+pie_plot(
+    inner_subtypes_df,
+    "Count",
+    "Node Size",
+    f"Distribution of Inner node children ({inner_threshold * 100:.1f}% threshold)",
+)
+
+# Bar: Inner node subtypes above 2% threshold
+inner_bar_threshold = 0.02
+inner_bar_df, _ = filter_and_group(df, "Inner", threshold=inner_bar_threshold)
+bar_plot(
+    inner_bar_df,
+    "Node Size",
+    "Count",
+    "Inner node children count",
+    order=inner_bar_df.sort_values("Count", ascending=False)["Node Size"],
+    pad=20,
+    rotation=45,
+)
```

Comment:
What is the purpose of these plots? Without the "other" I don't see how this data should be interpreted.